server/context.go

```diff
@@ -82,6 +86,10 @@ func (s *SessionManager) SetDB(conn *mysql.Conn, db string) error {
 		return err
 	}
 
+	if db != "" && !s.hasDBFunc(db) {
+		return fmt.Errorf("database '%s' not found", db)
```

Comment:
Should use sql.ErrDatabaseNotFound instead

Reply:
done